drivers/I2C.cpp

```diff
@@ -135,6 +140,53 @@ void I2C::unlock()
     _mutex->unlock();
 }
 
+int I2C::recover(PinName sda, PinName scl)
+{
+    DigitalInOut pin_sda(sda, PIN_INPUT, PullNone, 1);
+    DigitalInOut pin_scl(scl, PIN_INPUT, PullNone, 1);
+
+    // Read and verify if recovery is required
+    if (pin_scl == 1) {
+        if (pin_sda == 1) {
+            // Return successfuly as SDA and SCL is high
+            return 0;
+        }
+    } else {
+        // Return as SCL is low and no access to become master.
+        return I2C_ERROR_BUS_BUSY;
+    }
```

Comment:
This could be rearranged:
```cpp
// Return as SCL is low and no access to become master.
if (pin_scl == 0) {
  return I2C_ERROR_BUS_BUSY;
}

// Return successfully as SDA and SCL is high
if (pin_sda == 1) {
  return 0;
}
```

Reply:
Done